### PR TITLE
Sync document and entity PostManager

### PR DESCRIPTION
**Pull request by core23**
### Changelog

```markdown
### Fixed
 - Sync `PostManager` for document and entity repoitories
```

### Subject

Both managers have a different behavior. Some parts were only added to the entity repository but should exist in the document repository too. 

Renamed internal variables.

### CODE REVIEW

File: Document/PostManager.php [hunk 1]
```diff
@@ -96,32 +96,52 @@ public function findOneByPermalink($permalink, BlogInterface $blog)
...
+        if (!isset($criteria['enabled']) && $criteria['mode'] == 'public') {
+            $criteria['enabled'] = true;
+        }
+        if (isset($criteria['enabled'])) {
+            $query->andWhere('p.enabled = :enabled');
+            $parameters['enabled'] = $criteria['enabled'];
+        }
 
-        if (isset($criteria['date'])) {
+        if (isset($criteria['date']) && isset($criteria['date']['query']) && isset($criteria['date']['params'])) {
```
Comment (greg0ire) [inline Document/PostManager.php]: `isset` can take several arguments

File: Entity/PostManager.php [hunk 2]
```diff
@@ -30,65 +30,46 @@ class PostManager extends BaseEntityManager implements PostManagerInterface
...
-                $query->andWhere($pdqp['query']);
-            }
+        $parameters = array();
 
-            if (isset($urlParameters['slug'])) {
-                $query->andWhere('p.slug = :slug');
-                $parameters['slug'] = $urlParameters['slug'];
-            }
+        if (isset($urlParameters['year'], $urlParameters['month'], $urlParameters['day'])) {
+            $dateQueryParts = $this->getPublicationDateQueryParts(sprintf('%d-%d-%d', $urlParameters['year'], $urlParameters['month'], $urlParameters['day']), 'day');
```
Comment (greg0ire) [inline Entity/PostManager.php]: Please linebreak this

File: Document/PostManager.php (hunk 1 above)
Comment (Soullivaneuh) [inline Document/PostManager.php]: Would be awesome to have a cs fixer for that.

Comment (Soullivaneuh) [inline Document/PostManager.php]: Annnnnd new issue is here: https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/1977

**Comment by core23** [comment]
RTM @greg0ire ?

**Comment by greg0ire** [comment]
Yes! Thanks!

**Comment by OskarStark** [comment]
minor or patch @greg0ire ?

**Pull request merged**